docs/tutorial/add-subscription.md

```diff
@@ -0,0 +1,104 @@
+---
+id: add-subscription
+title: Add subscription
+sidebar_label: 9. Add subscription (advanced)
+---
+
+We discovered how to Query and Mutate datas, now, we are going to discover a most advanced part of GraphQL, the [event-based subscriptions](https://graphql.org/blog/subscriptions-in-graphql-and-relay/#event-based-subscriptions), well-known as `subscription` in the specification.
+
+In our tutorial, we will implement a subscription which will wait the end of the cooking process for a given recipe.
+
+During the cooking process, we will notify the user each second about the remaining time, and at the end, we will tell that the cooking time is over.
+
+This feature will allow the client of our GraphQL API to update in real time its UI with the remaining time.
+
+For development purpose, we consider that the cooking duration time is in second instead of minutes _(Our oven is really performant !)_.
+
+## **recipes_manager/sdl/Subscription.graphql**
+
+Fist, as usual, we define the SDL.
+
+* `launchAndWaitCookingTimer`: The client will subscribe to this field _(with the recipe id in parameter)_, and will receive a message each second which will respect the `CookingTimer` type.
+
+```graphql
+type Subscription {
+  launchAndWaitCookingTimer(id: Int!): CookingTimer
+}
+
+enum CookingStatus {
+  COOKING,
+  COOKED
+}
+
+type CookingTimer {
+  remainingTime: Int!,
+  status: CookingStatus!
+}
+```
+
+## **recipes_manager/subscription_resolvers.py**
+
+In the tartiflette engine, we implemented subscription simply by implementing a specific decorator _(@Subscription)_ over a function which returns generators, that it. For advanced use-cases, we suggest you to use pub/sub mecanism like Redis, Nats, Google Pub/Sub, Amazon EQS etc...
+
+```python
+import asyncio
+
+from tartiflette import Subscription
+
+from recipes_manager.data import INGREDIENTS_QUANTITY
+
+@Subscription("Subscription.launchAndWaitCookingTimer")
+async def subscription_cooking_time(
+    parent_result, args, ctx, info
+):
+  recipe = [r for r in RECIPES if r["id"] == int(args["id"])]
+
+  if not recipe:
+    raise Exception(f"The recipe with the id '{args['d']}' doesn't exist.")
+
+  for index in range(0, recipe[0]["cookingTime"]):
+    yield {
+      "remainingTime": recipe[0]["cookingTime"] - index,
+      "status": "COOKING"
+    }
+            
+    await asyncio.sleep(1)
+    
+  yield {
+    "remainingTime": 0,
+    "status": "COOKED"
+  }
+```
+
+## How could we use it?
+
+As we specified previously, for the tutorial, we decided to use the simplest form of the Subscription in Tartiflette, and we do not explained the transport layer.
```

Comment:
As specified previously, for this tutorial, we decided to use the simplest form of a Subscription in Tartiflette, and we don't cover the transport layer.